### ISSUE TITLE
watershed only takes shapefile as pour point input

### ISSUE THREAD

**Issue by lucfr** [jblindsay/whitebox-tools#12]
Which seems a bit inconsistent when necessary pre processing tools like snap pour points don't output shapefiles and raster to vector points is not yet ported.

**Comment by jblindsay** [comment]
@lucfr You're right that this is inconsistent. Early on, I had made the input of the Watershed tool take either a raster or a vector input. This either/or proved challenging to support, specifically, for the QGIS plugin for WhiteboxTools. Between supporting vector or raster pour point inputs, I think that vector inputs are probably more widely used and so I decided. The thinking for the snap pour points tools was always that I would update them to input/output vector as soon as I add support for writing vector files. In fact, I have just completed the code for writing shapefile geometries (attribute data are still a work in progress). I need to test this functionality by using it in a tool. The SnapPourPoint and JensenSnapPourPoint tools are the perfect candidates. As such, hopefully this inconsistency will be rectified very soon.
